### PR TITLE
Add onchain fees to funding and closing TX

### PR DESCRIPTION
**Pull request by omerfirmak**
As these are collaborative, they can use the channel funds for onchain fees.

Any fees associated with a non-collaborative closing attempt should be handled differently and paid by the peer that is attempting to close the channel.

### CODE REVIEW

**Comment by Geod24** [comment]
Any update on this ?

**Comment by omerfirmak** [comment]
> 
> 
> Any update on this ?

Will wrap this one up today.

**Comment by omerfirmak** [comment]
This is ready, but will probably need one or two similar sized PRs to close #2200 .

**Comment by omerfirmak** [comment]
Dropping https://github.com/bosagora/agora/pull/2227/commits/2b90a11119028d6dc061c9debbf30763e944a808 as that approach is not working out.

File: source/agora/flash/Scripts.d [hunk 1]
```diff
@@ -167,21 +167,30 @@ public Unlock createUnlockSettle (Signature sig, in ulong seq_id)
...
-    in Amount capacity, in Point pair_pk) @safe nothrow
+    in Amount capacity, in Point pair_pk, Amount fee) @safe nothrow
 {
     auto inputs = [Input(utxo_hash)];
     auto outputs = [Output(capacity,
         Lock(LockType.Key, pair_pk[].dup))];
 
+    // Account for the refund output
+    auto max_size = Transaction(inputs,
+        outputs ~ [Output(Amount.init, utxo.output.lock)]).sizeInBytes();
```
Comment (Geod24) [inline source/agora/flash/Scripts.d:186]: CC @kchulj , this seems to be needed frequently enough.

**Comment by omerfirmak** [comment]
Rebased

**Review by Geod24 — APPROVED** [review]

File: source/agora/flash/Scripts.d (hunk 1 above)
Comment (kchulj) [reply source/agora/flash/Scripts.d:186]: Hmm could make use of this in `TxBuilder`.

**Pull request merged**